apps/src/authoredHints.js

```diff
@@ -0,0 +1,211 @@
+/**
+ * @overview helper class to manage the state of the Authored Hint UI.
+ * Used exclusively by StudioApp.
+ */
+
+var dom = require('./dom');
+var HintSelect = require('./templates/hintSelect.jsx');
+var authoredHintUtils = require('./authoredHintUtils');
+var lightbulbSVG = require('./templates/lightbulb.svg.ejs')();
+var lightbulbDimSVG = require('./templates/lightbulb_dim.svg.ejs')();
+
+var AuthoredHints = function (studioApp) {
+  this.studioApp_ = studioApp;
+
+  /**
+   * @typedef {Object} AuthoredHint
+   * @property {string} content
+   * @property {string} hintId
+   * @property {string} hintClass
+   * @property {string} hintType
+   * @property {boolean} alreadySeen
+   */
+  /**
+   * @type {!AuthoredHint[]}
+   */
+  this.hints_ = [];
+
+  /**
+   * @type {number}
+   */
+  this.scrptId_ = undefined;
+
+  /**
+   * @type {number}
+   */
+  this.levelId_ = undefined;
+
+  /**
+   * @type {Element}
+   */
+  this.lightbulb = document.createElement('div');
+  this.lightbulb.id = "lightbulb";
+};
+
+module.exports = AuthoredHints;
+
+/**
+ * @return {AuthoredHints[]}
+ */
+AuthoredHints.prototype.getUnseenHints = function () {
+  return this.hints_ ? this.hints_.filter(function (hint) {
+    return hint.alreadySeen === false;
+  }) : [];
+};
```

Comment:
could also right this as:
```
return (this.hints_ || []).filter(....);
```
which might be marginally more easy to read (i have no strong opinion here)